### PR TITLE
[multiple] Update for externals changes + Ubuntu 24.04 externals builder + various improvements

### PR DESCRIPTION
**Pull request by SwooshyCueb**
Addresses #130
Addresses #136
Addresses #139
Addresses #140
Addresses #142
In service of irods/irods#7382
In service of irods/irods#7592
In service of irods/externals#236
In service of irods/externals#239
In service of irods/externals#244

Companion PRs:
  - irods/irods#7733
  - irods/irods#7739
  - irods/irods_client_icommands#477
  - irods/irods_client_icommands#478
  - irods/externals#246
  - irods/irods_testing_environment#210

### CODE REVIEW

**Comment by SwooshyCueb** [comment]
Ready for review

**Review by alanking — COMMENTED** [review]
Just had one question about catchsegv. Otherwise looks good to me

File: build_debuggers.ubuntu22.Dockerfile [hunk 1]
```diff
@@ -83,6 +83,17 @@ RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
     && \
     rm -rf /tmp/*
 
+#--------
+# catchsegv
```
[the remark this answers is not in the document]
Comment (SwooshyCueb) [reply build_debuggers.ubuntu22.Dockerfile:87]: It's not available on all platforms, as it was removed from glibc proper at some point not too long ago, and most distros don't package glibc-tools.

[the remark this answers is not in the document]
Comment (alanking) [reply build_debuggers.ubuntu22.Dockerfile:87]: Ah, okay. Thanks

**Review by alanking — APPROVED** [review]
Seems good to me... I think we can go ahead and # it.

**Comment by SwooshyCueb** [comment]
\#'d

**Pull request merged**